builder/src/upload.tsx

```diff
@@ -430,9 +453,26 @@ const SubmissionForm: React.FC<SubmissionFormProps> = observer((props) => {
                             progress={submissionProgress}
                         />
                     )}
-                    {formErrors.generalError && (
-                        <div className="px-3 py-3 text-danger field-errors">
-                            <p>{formErrors.generalError}</p>
+                    {hasErrors && (
+                        <div className="mb-0 px-3 py-3 alert alert-danger field-errors">
+                            <ul className="mx-0 my-0 pl-3">
+                                {formErrors.generalErrors.map((e, idx) => (
+                                    <li key={idx}>{e}</li>
```

Comment:
Not 100% sure but React might have some issues with sibling element having the same key, e.g. `0`, even though they're not from the same map-loop. Maybe add a error type based prefix to the keys?

Reply:
Good point, I've assumed those are internal to an individual `.map` call, but it'd make more sense it's the child element tree in its entirety. Will change just in case